imports/server/mediasoup-api.ts

```diff
@@ -0,0 +1,391 @@
+import Consumers from '../lib/models/mediasoup/consumers';
+import Peers from '../lib/models/mediasoup/peers';
+import ProducerClients from '../lib/models/mediasoup/producer_clients';
+import ProducerServers from '../lib/models/mediasoup/producer_servers';
+import Rooms from '../lib/models/mediasoup/rooms';
+import Routers from '../lib/models/mediasoup/routers';
+import TransportRequests from '../lib/models/mediasoup/transport_requests';
+import TransportStates from '../lib/models/mediasoup/transport_states';
+import Transports from '../lib/models/mediasoup/transports';
+import Servers from '../lib/models/servers';
+import { checkAdmin } from '../lib/permission_stubs';
+import { registerPeriodicCleanupHook, serverId } from './garbage-collection';
+import Locks from './models/lock';
+
+const ignoringDuplicateKeyErrors = (fn: () => void) => {
+  try {
+    fn();
+  } catch (e) {
+    // 11000 is a duplicate key error
+    if (!(e instanceof NpmModuleMongodb.MongoError) || e.code !== 11000) {
+      throw e;
+    }
+  }
+};
+
+registerPeriodicCleanupHook((deadServers) => {
+  Peers.remove({ createdServer: { $in: deadServers } });
+
+  // Deleting a room creates a potential inconsistency, since we might still
+  // have peers on other servers. Therefore, take out a lock to make sure we
+  // see a consistent view (and everyone else does too), then check if there
+  // are still peers joined to this room
+  Rooms.find({ routedServer: { $in: deadServers } }).forEach((room) => {
+    Locks.withLock(`mediasoup:room:${room.call}`, () => {
+      if (!Rooms.remove(room._id)) {
+        return;
+      }
+
+      const peer = Peers.findOne({ call: room.call });
+      if (peer) {
+        ignoringDuplicateKeyErrors(() => {
+          Rooms.insert({
+            call: room.call,
+            routedServer: serverId,
+            createdBy: peer.createdBy,
+          });
+        });
+      }
+    });
+  });
+});
+
+Meteor.publish('mediasoup:debug', function () {
+  if (!this.userId) {
+    throw new Meteor.Error(401, 'Not logged in');
+  }
+
+  checkAdmin(this.userId);
+
+  return [
+    Servers.find(),
+    Peers.find(),
+    Rooms.find(),
+    Routers.find(),
+    TransportRequests.find(),
+    Transports.find(),
+    TransportStates.find(),
+    ConnectRequests.find(),
+    ConnectAcks.find(),
+    ProducerClients.find(),
+    ProducerServers.find(),
+    Consumers.find(),
+    ConsumerAcks.find(),
+  ];
+});
+
+Meteor.publish('mediasoup:metadata', function (hunt, call) {
+  check(hunt, String);
+  check(call, String);
+
+  if (!this.userId) {
+    throw new Meteor.Error(401, 'Not logged in');
+  }
+
+  return Peers.find({ hunt, call });
+});
+
+Meteor.publish('mediasoup:join', function (hunt, call, tab) {
+  check(hunt, String);
+  check(call, String);
+  check(tab, String);
+
+  if (!this.userId) {
+    throw new Meteor.Error(401, 'Not logged in');
+  }
+
+  if (Flags.active('disable.webrtc')) {
+    throw new Meteor.Error(403, 'WebRTC disabled');
+  }
+
```

Comment:
For completeness: maybe we should check that user `this.userId` is a member of hunt `hunt` or throw here